enumerable_methods.rb

```diff
@@ -0,0 +1,217 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    ind = 0
+    for i in self
+      yield(i, ind)
+      ind += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << n if yield(n)
+    end
+    new_arr
+  end
+
+  def arg?(item, arg)
+    return item.is_a? arg if arg.is_a? Class
+    return !(item =~ arg).nil? if arg.is_a? Regexp
+
+    item == arg
+  end
+
+  def my_all?(arg = nil)
+    my_each do |item|
+      case block_given?
+      when false
+        (return false unless item) unless arg
+        (return false unless arg?(item, arg)) if arg
+      else
+        return false unless yield(item)
+      end
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    my_each do |item|
+      case block_given?
+      when false
+        return true if !arg && item
+        return true if arg && is_arg?(item, arg)
+      else
```

Comment:
your` is_arg?` is not defined on your file, so it raises an error. Please fix this.